pkg/build/api/validation/validation.go

```diff
@@ -103,11 +103,18 @@ func validateBuildSpec(spec *buildapi.BuildSpec, fldPath *field.Path) field.Erro
 	allErrs := field.ErrorList{}
 	s := spec.Strategy
 
-	if s.CustomStrategy == nil && spec.Source.Git == nil && spec.Source.Binary == nil && spec.Source.Dockerfile == nil {
+	if s.CustomStrategy == nil && s.JenkinsPipelineStrategy == nil && spec.Source.Git == nil && spec.Source.Binary == nil && spec.Source.Dockerfile == nil {
```

Comment:
>  If the Jenkinsfile is stored in the gitrepo, does it get reprocessed every time the build runs?

Yes. This is how Jenkins works for pipelines - it's the pipeline definition stored in the Git repo that defines the actual build steps.

> When does the jenkins Job definition get updated?

The job definition doesn't need to change if you're using a Jenkinsfile from a git repo. If you're using an inline Jekninsfile in BuildConfig, the job is updated automatically by our Jenkins plugin when the BC is changed. You can switch between git repo stored or inline & job is updated correctly so migration either way works nicely.

> what impact does that have if the job definition changes significantly between builds?

There's a difference between the Jenkins job definition (XML definition of the job) & the pipeline definition. Our job definition right now is pretty sparse, only allowing changes to the pipeline definition. The pipeline definition can't define all things (e.g. triggers) that are available on a generic job though & we're going to have to see how much we want to be able to specify at the job level. There is also the possibility of expanding this to support generic non-pipeline jobs but that introduces a lot more complexity that we need to see if it's worth adding/supporting.